tests/rustdoc-ui/infinite-recursive-type-impl-trait.rs

```diff
@@ -1,6 +1,7 @@
+pub enum E { //~ ERROR
```

Comment:
It isn't triggered anymore. Hence why I moved it. Should I just remove the test then?

Reply:
Changing this to check-pass seems ok (moving the item to the top-level was definitely incorrect, thank you @notriddle for catching it). tbh I was hesitant to add this test originally, given that there are many other similar ICEs that we haven't bothered fixing in 3 years, but if in the future something changes we can revisit the decision.

While you're looking at this, could you please rename the test to `tests/rustdoc-ui/error-in-impl-trait/infinite-recursive-type.rs` so it's easier to recover the context? It sounds like this change was because of `--document-private-items`, so if you could also add a revision of the test where `f` and `E` are both public, that would be helpful too.